### ISSUE TITLE
The cli should respect XDG spec for config and credential file storage

### ISSUE THREAD

**Issue by jasonkarns** [aws/aws-cli#9031]
The XDG spec defines where config files and credential files should be placed.

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Instead of writing to $HOME, the aws config directory should be `$XDG_CONFIG_HOME/aws`. (And `$XDG_DATA_HOME/aws/` for creds)

**Comment by zackheil** [comment]
It doesn't really matter to me that aws-cli doesn't support XDG out of the box, and I realize a request to conform an existing app to a specific metadata file configuration is a tall order, so I wouldn't even say to complete this you would need to go that far. I've found the comment further up mentions a standard method to reroute most of the areas the CLI touches to go a custom location via the environment variables: `AWS_CONFIG_FILE`, `AWS_CLI_HISTORY_FILE`, `AWS_CREDENTIALS_FILE`, `AWS_WEB_IDENTITY_TOKEN_FILE`, and `AWS_SHARED_CREDENTIALS_FILE`. The issue I have with this method (that blocks it from being a complete solution) is the fact that there are spots in code that hard-code where they are looking at without checking an environment variable. From a brief search of the code and from experiencing myself:

- https://github.com/aws/aws-cli/blob/develop/awscli/customizations/assumerole.py#L8
- https://github.com/aws/aws-cli/blob/develop/awscli/alias.py#L35 (has capability, but all instances that use this parameter seem to just be test files from what this website will show)
- The aws-sso directory/cache `~/.aws/sso/cache` (couldn't locate in code, most annoying for me)

If there are environment variable solutions that drive these locations, that is good enough for me. It's just a bit sad to see all related issues for this get funneled to this one from 2017 that has no traction at all

**Comment by TomaszGasior** [comment]
> It doesn't really matter to me that aws-cli doesn't support XDG out of the box

The whole point of the standards it to support them out of the box. User should not have to care about putting environment variables just to make software behave correctly.

**Comment by lousyd** [comment]
> It doesn't really matter to me that aws-cli doesn't support XDG out of the box

It does matter to me, and in fact, when they were working on v2 of the cli they asked for feedback on what people would like to see in v2. [I specifically submitted this issue](https://github.com/aws/aws-cli/issues/4662). That would have been the perfect opportunity to set this behavior but the suggestion was filed in the round file.

**Comment by nickjj** [comment]
> User should not have to care about putting environment variables just to make software behave correctly.

Yep and in this case based on reading through this thread, setting the environment variables doesn't solve the problem because there's a few paths that don't have environment variables or certain tools (Terraform, etc.) don't respect them.

I see this as an all or nothing type of thing. Either it all works and your home directory isn't polluted or it mostly works but you still have `~/.aws` lingering around. A partial solution seems worse since you have to define a bunch of env vars but you still have the issue in the end. In other words, this isn't solved by the current available options.

**Comment by hostaway-greg** [comment]
Any update on this? It will be 10 years soon ...

**Comment by devnoot** [comment]
> Any update on this? It will be 10 years soon ...

They're going for 20
